### PR TITLE
LPS-153155 Use fetch directly to avoid doing unneeded requests

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by victorg1991** [comment]
ci:test:sf

**Comment by victorg1991** [comment]
ci:test:relevant

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/AssetVocabularyCategoriesSelector.es.js [hunk 1]
```diff
@@ -41,37 +45,39 @@ function AssetVocabulariesCategoriesSelector({
...
+								'name': `%${inputValue.toLowerCase()}%`,
+								'start': 0,
+								'vocabularyIds': sourceItemsVocabularyIds,
+							},
+						}),
+						p_auth: Liferay.authToken,
+					}),
+					'credentials': 'include',
+					'method': 'POST',
+					'x-csrf-token': Liferay.authToken,
```
Comment (p2kmgcl) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/AssetVocabularyCategoriesSelector.es.js:74]: I've checked and frontend-js-web's fetch already includes "credentials" and "x-csrf-token" for local requests: https://github.com/liferay/liferay-portal/blob/0f5d6400bd0d70be4c09b26181684dd79a1d466b/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/fetch.es.js#L53-L56

:D

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/AssetVocabularyCategoriesSelector.es.js [hunk 2]
```diff
@@ -41,37 +45,39 @@ function AssetVocabulariesCategoriesSelector({
...
 
 	const previousInputValue = usePrevious(inputValue);
 
 	useEffect(() => {
 		if (inputValue && inputValue !== previousInputValue) {
-			refetch();
+			fetch(
+				`${
+					window.location.origin
+				}${themeDisplay.getPathContext()}/api/jsonws/invoke`,
```
Comment (p2kmgcl) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/AssetVocabularyCategoriesSelector.es.js:57]: Can't we write just `/api/jsonws/invoke` here?

**Review by p2kmgcl — COMMENTED** [review]
Awesome :smile: 

I added a couple of comments

**Comment by victorg1991** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:reevaluate:1317401_5198

**Comment by ozysouza** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ozysouza** [comment]
Just started reviewing :)

**Comment by ozysouza** [comment]
ci:close

**Pull request closed without merging**